src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -252,7 +253,53 @@ public static string LocalSlotName
 
             this.logger.LogInformation(
                 "{instanceId}: Function '{functionName} ({functionType})', version '{version}' received a '{eventName}' event. State: {state}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {extensionVersion}. SequenceNumber: {sequenceNumber}.",
-                instanceId, functionName, functionType, version, eventName, FunctionState.ExternalEventRaised, hubName, LocalAppName, LocalSlotName, ExtensionVersion, this.sequenceNumber++);
+                instanceId, functionName, functionType, version, eventName, FunctionState.ExternalEventRaised, hubName,
+                LocalAppName, LocalSlotName, ExtensionVersion, this.sequenceNumber++);
+        }
+
+        public void SendMessageFailed(
+            string hubName,
+            string functionName,
+            FunctionState functionState,
+            string version,
+            string instanceId,
+            HttpStatusCode statusCode,
+            string httpStatusCodeReasonPhrase,
+            string reason)
+        {
+            FunctionType functionType = FunctionType.Orchestrator;
+            bool isReplay = false;
+
+            EtwEventSource.Instance.ExternalEventRaised(
+                hubName,
+                LocalAppName,
+                LocalSlotName,
+                functionName,
+                functionState,
+                version,
+                instanceId,
+                statusCode,
+                httpStatusCodeReasonPhrase,
+                reason,
+                functionType,
+                ExtensionVersion,
+                isReplay);
+
+            this.logger.LogError(
+                "Error in sending message to the EventGrid. Please check the host.json configuration durableTask.EventGridTopicEndpoint and EventGridKey. LifeCycleNotificationHelper.SendNotificationAsync - Status: {statusCode} Reason Phrase: {httpStatusCodeReasonPhrase} For more detail: {instanceId}: Function '{functionName} ({functionType})', function state {functionState} version '{version}' failed with an error. Reason: {reason}. IsReplay: {isReplay}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {ExtensionVersion}.",
```

Comment:
There are several things I think we should change regarding this message. Here is a version that I like better:

```csharp
$"Failed to send the {{functionState}} notification to Azure Event Grid. Please check the configuration values for {nameof(EventGridTopicEndpoint)} and {nameof(EventGridKeySettingName)}. Status code: {{statusCode}}. Reason: {{reason}}. Function '{{functionName}} ({{functionType}})'. InstanceId: {{instanceId}}. Version '{{version}}'. IsReplay: {{isReplay}}. HubName: {{hubName}}. AppName: {{appName}}. SlotName: {{slotName}}. ExtensionVersion: {{ExtensionVersion}}."